lib/tools/node-modules/node-modules-dependencies-builder.ts

```diff
@@ -47,7 +47,18 @@ export class NodeModulesDependenciesBuilder implements INodeModulesDependenciesB
 			}
 		}
 
-		return resolvedDependencies;
+		return this.filterUniqueDependencies(resolvedDependencies);
+	}
+
+	private filterUniqueDependencies(dependencies: IDependencyData[]): IDependencyData[] {
+		const depsDictionary = dependencies.reduce((dict, dep) => {
+			const collision = dict[dep.name];
+			if (!collision || collision.depth > dep.depth) {
```

Comment:
I'm not sure I understand what's the idea here, can you please add comment in the code